src/test/regress/sql/multi_task_assignment_policy.sql

```diff
@@ -236,19 +235,31 @@ $cmd$, 'task_assignment_nonreplicated_hash');
 SELECT count(DISTINCT value) FROM explain_outputs;
 TRUNCATE explain_outputs;
 
-RESET citus.task_assignment_policy;
-RESET client_min_messages;
-
 -- we should be able to use round-robin with router queries that 
 -- only contains intermediate results
-BEGIN;
 CREATE TABLE task_assignment_test_table_2 (test_id integer);
 SELECT create_distributed_table('task_assignment_test_table_2', 'test_id');
 
-WITH q1 AS (SELECT * FROM task_assignment_test_table_2) SELECT * FROM q1;
-SET LOCAL citus.task_assignment_policy TO 'round-robin';
-WITH q1 AS (SELECT * FROM task_assignment_test_table_2) SELECT * FROM q1;
-ROLLBACK;
+SET citus.task_assignment_policy TO 'round-robin';
+
+-- run the query two times to make sure that it hits two different worrkers
+-- on consecutive runs
+INSERT INTO explain_outputs
+SELECT parse_explain_output($cmd$
+EXPLAIN WITH q1 AS (SELECT * FROM task_assignment_test_table_2) SELECT * FROM q1
```

Comment:
This does not test the code.
1 - it needs to be router query, this is not
2 - you need a replication factor > 1
3 - or test with a reference table

Reply:
The CTE here is replaced with an `intermediate_result` and is rewritten into:
```sql
SELECT test_id
FROM (
  SELECT intermediate_result.test_id
  FROM read_intermediate_result('13_1'::text, 'binary'::citus_copy_format)
    intermediate_result(test_id integer)) q1;
```

The resulting query is router executable, and the it is executed on different workers if using round-robin. Here are 2 `EXPLAIN` outputs:

```
                                                                 QUERY PLAN
--------------------------------------------------------------------------------------------------------------------------------------------
 Custom Scan (Citus Router)  (cost=0.00..0.00 rows=0 width=0)
   ->  Distributed Subplan 15_1
         ->  Custom Scan (Citus Real-Time)  (cost=0.00..0.00 rows=0 width=0)
               Task Count: 32
               Tasks Shown: One of 32
               ->  Task
                     Node: host=localhost port=9701 dbname=postgres
                     ->  Seq Scan on task_assignment_test_table_2_102043 task_assignment_test_table_2  (cost=0.00..35.50 rows=2550 width=4)
   Task Count: 1
   Tasks Shown: All
   ->  Task
         Node: host=localhost port=9702 dbname=postgres
         ->  Function Scan on read_intermediate_result intermediate_result  (cost=0.00..10.00 rows=1000 width=4)
(13 rows)
```
```
                                                                 QUERY PLAN
--------------------------------------------------------------------------------------------------------------------------------------------
 Custom Scan (Citus Router)  (cost=0.00..0.00 rows=0 width=0)
   ->  Distributed Subplan 17_1
         ->  Custom Scan (Citus Real-Time)  (cost=0.00..0.00 rows=0 width=0)
               Task Count: 32
               Tasks Shown: One of 32
               ->  Task
                     Node: host=localhost port=9701 dbname=postgres
                     ->  Seq Scan on task_assignment_test_table_2_102043 task_assignment_test_table_2  (cost=0.00..35.50 rows=2550 width=4)
   Task Count: 1
   Tasks Shown: All
   ->  Task
         Node: host=localhost port=9701 dbname=postgres
         ->  Function Scan on read_intermediate_result intermediate_result  (cost=0.00..10.00 rows=1000 width=4)
(13 rows)
```

I should have explained this better in the comments. I may also add a simpler router query in this test to make it easier to read&understand